src/graphics.rs

```diff
@@ -1628,6 +1634,263 @@ impl Drawable for Sprite {
+        println!("This many chars: {}", char_list.len());
+        let chars_per_row = ((char_list.len() as f32).sqrt() as usize) + 1;
+        let bytes_font = font.bytes().collect::<Result<Vec<u8>, IoError>>()?;
+        let rusttype_font = ::rusttype::Font::try_from_bytes(&bytes_font[..]).ok_or(anyhow!(
+            "Unable to create a rusttype::Font using bytes_font"
+        ))?;
+        let mut glyphs_and_chars = char_list
+            .iter()
+            .map(|c| {
+                (
+                    rusttype_font
+                        .glyph(*c)
+                        .scaled(::rusttype::Scale {
+                            x: font_size,
+                            y: font_size,
+                        })
+                        .positioned(::rusttype::Point { x: 0.0, y: 0.0 }),
+                    *c,
+                )
+            })
+            .collect::<Vec<(::rusttype::PositionedGlyph, char)>>();
+        glyphs_and_chars
+            .sort_unstable_by_key(|g| g.0.pixel_bounding_box().unwrap_or(inval_bb).height());
+
+        let mut texture_height: usize = glyphs_and_chars
+            .last()
+            .unwrap()
+            .0
+            .pixel_bounding_box()
+            .unwrap_or(inval_bb)
+            .height() as usize;
+        let mut current_row: usize = 0;
+        let mut widest_row: usize = 0;
+        let mut row_sum: usize = 0;
+
+        // Sort the glyphs by height so that we know how tall each row should be in the atlas
+        // Sums all the widths and heights of the bounding boxes so we know how large the atlas will be
+        let mut char_rows = Vec::new();
+        let mut cur_row = Vec::with_capacity(chars_per_row);
+
+        for (glyph, c) in glyphs_and_chars.iter().rev() {
+            let bb = glyph.pixel_bounding_box().unwrap_or(inval_bb);
+
+            if current_row > chars_per_row {
+                current_row = 0;
+                texture_height += bb.height() as usize;
+                if row_sum > widest_row {
+                    widest_row = row_sum;
+                }
+                row_sum = 0;
+                char_rows.push(cur_row.clone());
+                cur_row.clear();
+            }
+
+            cur_row.push((glyph, *c));
+            row_sum += bb.width() as usize;
+            current_row += 1;
+        }
+        // Push remaining chars
+        char_rows.push(cur_row);
+
+        let texture_width = widest_row + (chars_per_row * MARGIN);
+        texture_height += chars_per_row * MARGIN;
+
+        let mut texture = RgbaImage::new(texture_width as u32, texture_height as u32);
+
+        let mut texture_cursor: (usize, usize) = (0, 0);
+
+        let mut char_map: HashMap<char, CharInfo> = HashMap::new();
+
+        for row in char_rows {
+            let height = row
+                .first()
+                .unwrap()
+                .0
+                .pixel_bounding_box()
+                .unwrap_or(inval_bb)
+                .height() as usize;
```

Comment:
Break this up into two `let`s, and it'll become more readable and also fewer lines. Maybe `let first_glyph = row.first().unwrap().0;` and then `let height = first_glyph.pixel_bounding_box().unwrap_or(inval_bb).height() as usize;`?